experimental/juvix/contract.jvx

```diff
@@ -0,0 +1,198 @@
+Map = Data.Map
+
+-- common module interfaces for data structures in Common?
+MonoidMeasure : Common.Measure
+MonoidMeasure =
+  type T = Nat
+
+  -- (<>) is normal semi-group  type class function
+  (<>)    = (<>)
+  measure = id
+
+
+-- need OrdMap, as modules that one goes to can't be modules?
+Accounts = Map.OrdMap Data.String MonoidMeasure
+
+-- we send in the string module, as it has a total order
+-- function named compare, that is also part of a general type class
+Token =
+  type Address  = s : String.T {String.length s == 36}
+
+  type Storage = {
+    total-supply : Nat.T;
```

Comment:
Are we going to allow `-` in identifiers? Will we be very permissive (like Agda) or more restrictive?

Reply:
I think we should be restrictive, but allow `-` as I personally find it more readable than camlCase or snake_case

Reply:
Hmm, I don't have an opinion on the case question but I also don't think we should be too restrictive, except as prohibited by the parser, we can always add conventions later.